src/scaladoc/scala/tools/nsc/doc/html/Doclet.scala

```diff
@@ -12,9 +12,17 @@ import doclet._
 /** The default doclet used by the scaladoc command line tool
   * when no user-provided doclet is provided. */
 class Doclet extends Generator with Universer {
+  import scala.reflect.internal.Reporter
 
-  def generateImpl() {
-    new html.HtmlFactory(universe, new ScalaDocReporter(universe.settings)).generate()
```

Comment:
When creating a `Doclet` the user would get a different reporter than the one specified when creating a `DocFactory`, this caused incorrect behaviour when printing the summary messages on the external reporter (i.e. no messages).

Since we're not forcing the user to create a `DocFactory` with a`ScalaDocReporter`, I thought it best to change the members to `Reporter` instead. WDYT?